src/nevermined/api/RegistryBaseApi.ts

```diff
@@ -219,69 +213,71 @@ export abstract class RegistryBaseApi extends Instantiable {
       observer.next(CreateProgressStep.RegisteringDid)
 
       // On-chain asset registration
-      if (nftAttributes) {
-        this.logger.log('Registering Mintable Asset', ddo.id)
-
-        const nftAttributesWithoutRoyalties = { ...nftAttributes, royaltyAttributes: undefined }
-
-        if (nftAttributes.ercType === 721) {
-          await didRegistry.registerMintableDID721(
-            didSeed,
-            nftAttributes.nftContractAddress,
-            checksum,
-            assetAttributes.providers || [this.config.neverminedNodeAddress],
-            publisher.getId(),
-            nftAttributesWithoutRoyalties,
-            serviceEndpoint,
-            ddoVersion.immutableUrl,
-            DEFAULT_REGISTRATION_ACTIVITY_ID,
-            txParams,
-          )
+      if (publicationOptions.did != PublishOnChainOptions.OnlyOffchain) {
+        if (nftAttributes) {
+          this.logger.log('Registering Mintable Asset', ddo.id)
+
+          const nftAttributesWithoutRoyalties = { ...nftAttributes, royaltyAttributes: undefined }
+
+          if (nftAttributes.ercType === 721) {
+            await didRegistry.registerMintableDID721(
```

Comment:
Does make sense to call the observer depending of the cases in order to get deeper logging of what it is happening

Reply:
I think the observer is already registering what's going on. Is there any "event" that you miss?